File: perl/examples/lvs.pl
#!/usr/bin/perl -w

use strict;

use Sys::Guestfs;

# Look for LVM LVs, VGs and PVs in a guest image.

die "Usage: lvs.pl guest.img\n" if @ARGV != 1 || ! -f $ARGV[0];

print "Creating the libguestfs handle\n";
my $h = Sys::Guestfs->new ();
$h->add_drive ($ARGV[0]);

print "Launching, this can take a few seconds\n";
$h->launch ();
$h->wait_ready ();

print "Looking for PVs on the disk image\n";
my @pvs = $h->pvs ();
print "PVs found: (", join (", ", @pvs), ")\n";

print "Looking for VGs on the disk image\n";
my @vgs = $h->vgs ();
print "VGs found: (", join (", ", @vgs), ")\n";

print "Looking for LVs on the disk image\n";
my @lvs = $h->lvs ();
print "LVs found: (", join (", ", @lvs), ")\n";

File: ocaml/examples/lvs.ml
(* An example using the OCaml bindings. *)

open Printf

let () =
  if Array.length Sys.argv <= 1 || not (Sys.file_exists Sys.argv.(1)) then (
    eprintf "Usage: lvs guest.img\n";
    exit 1
  );

  let h = Guestfs.create () in
  Guestfs.add_drive_opts h ~format:"raw" Sys.argv.(1);
  Guestfs.launch h;

  let pvs = Guestfs.pvs h in
  printf "PVs found: [ %s ]\n" (String.concat "; " (Array.to_list pvs));

  let vgs = Guestfs.vgs h in
  printf "VGs found: [ %s ]\n" (String.concat "; " (Array.to_list vgs));

  let lvs = Guestfs.lvs h in
  printf "LVs found: [ %s ]\n" (String.concat "; " (Array.to_list lvs));

  (* Helps to find any allocation errors. *)
  Gc.compact ()
